cmd/solution/name.go

```diff
@@ -49,14 +49,26 @@ func getSolutionNameFromArgs(cmd *cobra.Command, args []string, flagName string)
 		}
 		// We only want to append .dev for subscribing/unsubscribing commands
 		if solutionTag != "" && solutionTag != "stable" && (commandName == "subscribe" || commandName == "unsubscribe") {
-			name = fmt.Sprintf("%s.%s", name, solutionTag)
+			if solutionTag == "dev" {
+				name = fmt.Sprintf("%s.%s", name, solutionTag)
+			} else {
+				name = fmt.Sprintf("%s%s.dev", name, solutionTag)
+			}
 		}
 		return name
 	}
 
 	// return the solution name from flag, if provided
 	if nameFromFlag != "" {
-		return nameFromFlag
```

Comment:
please either refactor so that the logic is in one place (no duplicating) or leave the nameFromFlag unchanged, as it was used only for backward compatibility from before solution isolation was added; there should be a deprecation warning.